src/external_data_ifaces_impl.cpp

```diff
@@ -41,4 +42,48 @@ sdbusplus::async::task<> ExternalDataIFacesImpl::fetchBMCRedundancyMgrProps()
     co_return;
 }
 
+// NOLINTNEXTLINE
+sdbusplus::async::task<> ExternalDataIFacesImpl::fetchSiblingBmcIP()
+{
+    // TODO: Currently, the IP addresses for both BMCs are hardcoded.
+    // Once the network DBus exposes the IPs, update the logic to retrieve
+    // them dynamically from Dbus.
+
+    // TODO: Handle the exception and exit gracefully, as the data sync relies
+    //      heavily on these DBus properties and cannot function effectively
+    //      without them.
+
+    using SiblingBMCMgr = sdbusplus::client::xyz::openbmc_project::state::bmc::
+        redundancy::Sibling<>;
+
+    using SiblingBMC = sdbusplus::common::xyz::openbmc_project::state::bmc::
+        redundancy::Sibling;
+
+    std::string siblingBMCInstancePath =
+        std::string(sdbusplus::common::xyz::openbmc_project::state::bmc::
+                        redundancy::Sibling::namespace_path::value) +
+        "/" +
+        sdbusplus::common::xyz::openbmc_project::state::bmc::redundancy::
+            Sibling::namespace_path::bmc;
+
+    auto siblingBMCMgr = SiblingBMCMgr(_ctx)
+                             .service(SiblingBMC::interface)
+                             .path(siblingBMCInstancePath);
+
+    auto siblingBMCPosition = co_await siblingBMCMgr.bmc_position();
+
+    if (siblingBMCPosition == 0)
+    {
+        // Using the simics bmc0 eth0 IP address
+        siblingBmcIP("10.0.2.100");
+    }
+    else
+    {
+        // Using the simics bmc0 eth0 IP address
```

Comment:
> // Using the simics bmc0 eth0 IP address

// Using the simics bmc1 eth0 IP address

Reply:
Its not addressed yet.

Reply:
Oh, I see you addressed it in the next commit. That's fine for now, but please make sure to address review comments within their respective commits in the future.